rollup.config.js

```diff
@@ -2,25 +2,36 @@ import path from 'path'
 import babel from '@rollup/plugin-babel'
 import resolve from '@rollup/plugin-node-resolve'
 import typescript from '@rollup/plugin-typescript'
+import esbuild from 'rollup-plugin-esbuild'
 import { sizeSnapshot } from 'rollup-plugin-size-snapshot'
 
 const createBabelConfig = require('./babel.config')
-
-const { root } = path.parse(process.cwd())
-const external = (id) => !id.startsWith('.') && !id.startsWith(root)
 const extensions = ['.js', '.ts', '.tsx']
-const getBabelOptions = (targets) => {
+const { root } = path.parse(process.cwd())
+
+function external(id) {
+  return (
+    id.startsWith('./vanilla') || (!id.startsWith('.') && !id.startsWith(root))
+  )
+}
+
+function getEsbuild(target) {
+  return esbuild({
+    minify: false,
+    target,
+    tsconfig: path.resolve('./tsconfig.json'),
+  })
+}
+
+function getBabelOptions(targets) {
   const config = createBabelConfig({ env: (env) => env === 'build' }, targets)
   if (targets.ie) {
-    config.plugins = [
-      ...config.plugins,
-      '@babel/plugin-transform-regenerator',
-      ['@babel/plugin-transform-runtime', { helpers: true, regenerator: true }],
-    ]
-    config.babelHelpers = 'runtime'
+    config.plugins = [...config.plugins, '@babel/plugin-transform-regenerator']
+    config.babelHelpers = 'bundled'
```

Comment:
Let me check how much this would affect the bundle size.

Reply:
| | `'runtime'` | `'bundled'` | diff |
|---|---|---|---|
|index.js|3760|3561|-199|
|middleware.js|8720|9147|+427|
|shallow.js|620|620|0|
|vanilla.js|1798|1798|0|

Okay, this is more or less what I expected.
We will probably be refactoring `persist` in middleware.ts without async/await, so the diff is likely to be reduced in the future.
Let's go with bundled as cjs build is kind of fallback.